compiler/qsc/src/codegen/tests.rs

```diff
@@ -625,8 +625,8 @@ mod adaptive_ri_profile {
             block_2:
               br label %block_3
             block_3:
-              %var_2 = phi i64 [0, %block_1], [1, %block_2]
-              call void @__quantum__rt__integer_record_output(i64 %var_2, i8* null)
+              %var_3 = phi i64 [0, %block_1], [1, %block_2]
```

Comment:
why isn't this `%var_2`?

Reply:
Part of the bug fix is to start the index for inserted phi node variables at the highest variable including stores, rather than without including stores. I think this is needed to play nice with the fix for properly mapping mutable variables that get stored to other mutable variables, but there is one other approach I could try to see if it maintains both the reuse logic and fix for mapping.

Reply:
Yeah, trying to reuse variable IDs during mapping for phi nodes in SSA creates a cycle in the variable mapping that will cause this test case to hang. The bug is meant to fix scenarios where a store instruction uses a variable that itself was created via a store instruction (see the test case `ssa_transform_maps_store_instrs_that_use_values_from_other_store_instrs` in the ssa_transform/tests.rs file). This was the case that motivated the update that checks if the variable we are mapping to has it's own mapping and repeating that search until we find the actual source value. But if variable IDs are reused, it will overlap with other mappings that may not have been applied yet. I can't remove the reused variable from the map otherwise it will leave behind orphaned variables, but I can't skip putting variables created for phi into the map either because then those uses won't be updated to use the phi created value. So I changed the logic to not reuse variable IDs, which leaves the gap `var_0`, `var_1`, `var_3`.

If this becomes an issue for understanding the QIR later we could add another pass that tries to consolidate variable IDs, but since these are named `%var_0` instead of unnamed `%0` variables, LLVM doesn't care what they are called and does not force them to be sequential.